interfaces/builtin/socketcan.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const socketCanSummary = `allows use of SocketCAN network interfaces`
+
+const socketCanBaseDeclarationSlots = `
+  socketcan:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
I feel like your instinct here to disallow `socketcan` is correct, but for completeness of the review, why should the use of CAN networking be denied by default? (Again, I think I agree with you when imagining snaps in automotive/etc)

Reply:
Given CAN is often used in industrial and automotive applications (e.g. passenger lift controls, driving assistance systems) and these could have a role in ensuring human safety, I feel it is better to be conservative and allow store owners or device administrators to grant access to the resource in this case.